megamek/src/megamek/common/pathfinder/BoardClusterTracker.java

```diff
@@ -365,7 +366,13 @@ private boolean buildingPlowThroughRequired(Entity entity, MovementType relevant
         if (!hex.containsTerrain(Terrains.BLDG_CF) && !hex.containsExit(Terrains.FUEL_TANK_CF)) {
             return false;
         } else if (relevantMovementType == MovementType.Walker) {
-            int buildingCF = board.getBuildingAt(coords).getCurrentCF(coords);
+            Building building = board.getBuildingAt(coords);
```

Comment:
Could make this final

Reply:
Does that help performance at all, or is it just a style preference?

Reply:
The compiler might be able to improve the performance and it notes the variable won't change after being set.

Reply:
The perf benefits only apply to values which can be statically known (e.g. effective constants), as in that case results can be pre-computed, memoized, or omitted. There exists the possibility the JVM could elide a barrier when reading from a _final class member_, but my understanding is that is not implemented. In all other cases it is a nice-to-have to signal intent to other programmers.

Consider for local variables that if the compiler can determine if you violate `final`, it also knows if it is _effectively_ `final`.